bin/utest/pipe.c

```diff
@@ -0,0 +1,91 @@
+#include <assert.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <signal.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "utest.h"
+#include "util.h"
+
+static sig_atomic_t parent_signal_delivered = 0;
+static sig_atomic_t child_signal_delivered = 0;
+
+void parent_sigpipe_handler(int signo) {
+  parent_signal_delivered = 1;
+  return;
+}
+
+void child_sigpipe_handler(int signo) {
+  child_signal_delivered = 1;
+  return;
+}
```

Comment:
You can use one function `sigpipe_handler` and reset value of variable `signal_delivered` when you want to use it.